src/main/java/com/epam/internship/carrental/car/CarController.java

```diff
@@ -0,0 +1,294 @@
+                                                              @RequestHeader("Authorization") final String authorization) {
+        return carService.getCarByIdWithAuthorization(carId, authorization);
+    }
+
+    /**
+     * The method is used for inserting a new Car object into the database, allowed for users with authorization.
+     * The car is defined in the body of the request.
+     * <pre>
+     *     Method PUT
+     *     URL /api/v1/car
+     * </pre>
+     * Sample call /api/v1/car
+     *
+     * @param car           insertable Car object
+     * @param authorization authorization token from the header of the request
+     * @return ResponseEntity which contains the inserted Car
+     * and Response Code 200 on success, or Response Code 403 if the token doesn't match.
+     */
+    @PutMapping(path = "/car", consumes = "application/json")
+    public @ResponseBody
+    ResponseEntity<Car> insertNewCarWithAuthorization(@RequestBody final Car car,
+                                                      @RequestHeader("Authorization") final String authorization) {
+        return carService.insertNewCarWithAuthorization(car, authorization);
+    }
+
+    /**
+     * Updates a single Car object in the database, defined by the ID in the request parameter with the parameters from
+     * the request body.
+     * <pre>
+     *     Method POST
+     *     URL /api/v1/car/{carId}
+     * </pre>
+     * Sample call /api/v1/car/1
+     *
+     * @param carId         updateable car's ID
+     * @param newCarParams  updateable parameters
+     * @param authorization authorization token from the header of the request
+     * @return ResponseEntity which contains the updated Car
+     * and Response Code 200 on success, or Response Code 403 if the token doesn't match,
+     * or if there is no car with given ID.
+     */
+    @PostMapping(path = "/updatecar/{carId}", consumes = "application/json")
+    public @ResponseBody
+    ResponseEntity<Car> updateCarByGivenParametersWithAuthorization(@PathVariable final Long carId,
+                                                                    @RequestBody final Car newCarParams,
+                                                                    @RequestHeader("Authorization") final String authorization) {
+        return carService.updateCarByGivenParametersWithAuthorization(carId, newCarParams, authorization);
+    }
+
+    /**
+     * Returns all available cars, in a pageable format.
+     * <pre>
+     *     Method GET
+     *     URL /api/v1/free
+     * </pre>
+     * Sample call /api/v1/free?page=0
+     *
+     * @param pageable standard paging parameters in the request
+     * @return ResponseEntity which contains an Pageable list of all free cars
+     * and Response Code 200 on success.
+     */
+    @GetMapping(path = "/free")
+    public @ResponseBody
+    ResponseEntity<Page<Car>> getAllFreeCars(@PageableDefault final Pageable pageable) {
```

Comment:
The search service should deal with this too. A flag should indicate in the search method if the user wants to lists the free cars.